docs/UserGuide.md

```diff
@@ -135,6 +135,62 @@ If `NAME_STRING` = `John p/ Tom`, the command keyed in could look like `radd p/[
 This will create the presence of 2 phone number parameters. In such a case, the latter value will be taken. 
 As `Tom` is not a valid phone number, it will be rejected.
 
+#### Interpreting errors
+
+In most cases, the error displayed should be relatively self-explanatory. This sub-section aims to clear up any ambiguity 
+you might encounter when dealing with invalid commands or errors. 
+
+We will break our explanation down into the following categories:
+- Incorrect formats
+- Parameters do not conform to required values
+- Commands that require an [`INDEX`] (e.g `redit 1 n/John`)
+
+##### Incorrect formats
+This error typically shows a message along the following lines: `Invalid command format! [Information on how to use
+the command]`
+
+![Invalid command format](images/error-invalid-cmd-format.png)
+
+Typically this happens because the command does not match the required format. Check if you are missing any parameters.
+A special case of this takes place when the `INDEX` value is not a number.
+
+##### Parameters do not conform to required values
+This error typically shows a message along the following lines: `[Parameter name] must be [some condition]`. E.g 
```

Comment:
Should `E.g` be `E.g.`?

Saw this in other places. If you make the change, consider doing a search for `e.g`

Reply:
Done.